src/control/lib/support/log.go

```diff
@@ -0,0 +1,733 @@
+	}
+
+	return true, nil
+}
+
+// Get the server config from the running daos engine process
+func getRunningConf(log logging.Logger) (string, error) {
+	running_config := ""
+	runState, err := checkEngineState(log)
+	if err != nil {
+		return "", err
+	}
+
+	if runState {
+		cmd := "ps -eo args | grep daos_engine | head -n 1 | grep -oP '(?<=-d )[^ ]*'"
+		stdout, err := exec.Command("bash", "-c", cmd).Output()
+		if err != nil {
+			return "", errors.Wrap(err, "daos_engine is not running on server")
+		}
+		running_config = filepath.Join(strings.TrimSpace(string(stdout)), config.ConfigOut)
+	}
+
+	return running_config, nil
+}
+
+// Get the server config, either from the running daos engine or default
+func getServerConf(log logging.Logger, opts ...CollectLogsParams) (string, error) {
+	cfgPath, err := getRunningConf(log)
+
+	if cfgPath == "" {
+		cfgPath = filepath.Join(config.DefaultServer().SocketDir, config.ConfigOut)
+	}
+
+	if err != nil {
+		return cfgPath, nil
+	}
+
+	log.Debugf(" -- Server Config File is %s", cfgPath)
+	return cfgPath, nil
+}
+
+// Copy file from source to destination
+func cpLogFile(src, dst string, log logging.Logger) error {
+	log_file_name := filepath.Base(src)
+	log.Debugf(" -- Copy File %s to %s\n", log_file_name, dst)
+
+	err := common.CpFile(src, filepath.Join(dst, log_file_name))
+	if err != nil {
+		return errors.Wrap(err, "unable to Copy File")
+	}
+
+	return nil
+}
+
+// Copy Command output to the file
+func cpOutputToFile(target string, log logging.Logger, cp ...logCopy) (string, error) {
+	// Run command and copy output to the file
+	// executing as sub shell enables pipes in cmd string
+	runCmd := strings.Join([]string{cp[0].cmd, cp[0].option}, " ")
+	out, err := exec.Command("sh", "-c", runCmd).CombinedOutput()
+	if err != nil {
+		return "", errors.New(string(out))
+	}
+
+	cmd := strings.ReplaceAll(cp[0].cmd, " -", "_")
+	cmd = strings.ReplaceAll(cmd, " ", "_")
+	log.Debugf("Collecting DAOS command output = %s > %s ", runCmd, filepath.Join(target, cmd))
+
+	if err := ioutil.WriteFile(filepath.Join(target, cmd), out, 0644); err != nil {
+		return "", errors.Wrapf(err, "failed to write %s", filepath.Join(target, cmd))
+	}
+
+	return string(out), nil
+}
+
+// Create the Archive of log folder.
+func ArchiveLogs(log logging.Logger, opts ...CollectLogsParams) error {
+	var buf bytes.Buffer
+	err := common.FolderCompress(opts[0].TargetFolder, &buf)
+	if err != nil {
+		return err
+	}
+
+	// write to the the .tar.gz
+	tarFileName := fmt.Sprintf("%s.tar.gz", opts[0].TargetFolder)
+	log.Debugf("Archiving the log folder %s", tarFileName)
+	fileToWrite, err := os.OpenFile(tarFileName, os.O_CREATE|os.O_RDWR, os.FileMode(0755))
+	if err != nil {
+		return err
+	}
+	defer fileToWrite.Close()
+
+	_, err = io.Copy(fileToWrite, &buf)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Get the system hostname
+func GetHostName() (string, error) {
```

Comment:
not blocking on this, but there is a Go function `os.Hostname()` that can fetch the hostname without calling out to the shell.

Reply:
Thanks for review. I will test and updated when will work on second iteration. 

If I recall, I was trying to use that os.Hostname() but it was giving the full name including the domain name. So another option is to use the function and remove the domain from it.